### DISCUSSION TITLE
OSError: accelerometer not found

### DISCUSSION THREAD

**Discussion by guyford94** [micropython/micropython#9274] [Hardware & Peripherals]
Hey
I use pyboard:
(sysname='pyboard', nodename='pyboard', release='1.11.0', version='v1.11 on 2020-11-26', machine='VCC-GND(R)pyboard412-12m with STM32F412RE' )
I'm trying to run the simplest program for the accelerometer
"""
from pyb import *
accel = pyb.Accel()
accel.x()
"""
and get the error:
MPY: sync file systems
MPY: soft reboot
Traceback (most recent call last):
   File "main.py", line 5, in <module>
OSError: accelerometer not found
pyMCU v1.11 on 2020-11-26; VCC-GND(R)pyboard412-12m with STM32F412RE
Type "help()" for more information.

Am I missing something?

**Comment by robert-hh** [comment]
Which type of Pyboard do you use? It may not have the accelerometer at the board. And besides, your firmware version is quite old.

**Reply by guyford94** [reply]
Hey
I use pyboard:
(sysname='pyboard', nodename='pyboard', release='1.11.0', version='v1.11 on 2020-11-26', machine='VCC-GND(R)pyboard412-12m with STM32F412RE' )
The component is installed on the board

![WhatsApp Image 2022-09-11 at 15 27 41](https://user-images.githubusercontent.com/52482940/189527603-39257c90-0313-420f-8e96-14d3d52924b6.jpeg)

**Reply by dpgeorge** [reply]
That's not an official pyboard.

**Reply by robert-hh** [reply]
I asked because I did not find a product page at the web. The board layout seems similar to a genuine Pyboard V1.1, with some extensions. Your test snippet works on a genuine Pyboard. I would try to run the official Pyboard firmware first. If VCC-GND made a proper clone, it may work. If you have access to the VCC-GND firmware, you can always roll back.

**Reply by robert-hh** [reply]
I see that you have something connected to X10, which is the SDA line of I2C(1). The PyBoard documentation tells, that the I2C(1) bus (at X9 and X10) must only be used for I2C (if at all) when you want to uses accel. So please disconnect what you have connected to X10. If it is as serial interface, use another one.

**Reply by guyford94** [reply]
Hi, thanks for the answers
It's definitely not an original PYBOARD, but I managed to run the accelerometer commands on it.
A component - JETSON NANO - is indeed connected to the board, which communicates with the board via I2C communication.
Are you saying that it is not possible to activate the accelerometer when the board is connected and communicating in I2C with another component?

**Reply by jimmo** [reply]
@guyford94 Please remember that sales of (official) pyboards are part of what funds the development of this project (including this forum).

What I can say though is that there is no problem using additional devices I2C on the same bus as the accelerometer (i.e. "X" or I2C1) and this works on my pyboard. I suspect whatever other device you have connected is interfering with the bus in some way. Like @robert-hh says, you cannot use the pins for any other purpose than I2C of course.
